overlord/overlord.go

```diff
@@ -402,17 +402,28 @@ func (o *Overlord) requestRestart(t state.RestartType) {
 	}
 }
 
+var exitWithError = func(err error) {
+	logger.Noticef("exiting overlord: %s", err)
```

Comment:
not entirely sure we want to use logging here,  cmd/snapd/maing.go uses `fmt.Fprintf(os.Stderr, "cannot run daemon: %v\n", err)`,  maybe here we should use:

```
fmt.Fprintf(os.Stderr, "cannot preseed %v\n", err)
```